utility_analysis_new/parameter_tuning.py

```diff
@@ -80,3 +84,148 @@ def sort_histogram(bins):
+            if isinstance(value, int):
+                if value <= 0:
+                    raise ValueError(f"{name} must be >0, but {name}={value}")
+            elif isinstance(value, Bounds):
+                if value.lower <= 0:
+                    raise ValueError(
+                        f"{name} lower bound must be >0, but {name}.lower={value}"
+                    )
+
+        check_int_attribute(self.max_partitions_contributed,
+                            "max_partitions_contributed")
+        check_int_attribute(self.max_contributions_per_partition,
+                            "max_contributions_per_partition")
+
+
+@dataclass
+class TuneOptions:
+    """Options for tuning process.
+
+    Note that parameters that are not tuned (e.g. metrics, noise kind) are taken
+    from aggregate_params.
+
+    Attributes:
+        epsilon, delta: differential private budget for aggregations for which
+          tuning is performed.
+        aggregate_params: parameters of aggregation.
+        function_to_minimize: which function of the error to minimize. In case
+          if this argument is a callable, it should take 1 argument of type
+          AggregateErrorMetrics and return float.
+        parameters_to_tune: spefies which parameters are tunable and what
+          restrictions on their values.
+    """
+    epsilon: float
+    delta: float
+    aggregate_params: pipeline_dp.AggregateParams
+    function_to_minimize: Union[MinimizingFunction, Callable]
+    parameters_to_tune: TunedParameters
+
+
+@dataclass
+class TuneResult:
+    """Represents tune results.
+
+    Attributes:
+        recommended_params: recommended parameters to use, according to
+          minimizing function. Note, that those parameters might be not optimal,
+          since finding the optimal parameters not always feasible.
+        options: input options of tuning (for convenience).
+        contribution_histograms: histograms of privacy id contributions.
+        utility_analysis_runs: the results of all utility analysis runs that
+          were performed during tuning process.
+    """
+    recommended_params: pipeline_dp.AggregateParams
+    options: TuneOptions
+    contribution_histograms: ContributionHistograms
+    utility_analysis_runs: List[UtilityAnalysisRun]
+
+
+def tune_parameters(col,
+                    contribution_histograms: ContributionHistograms,
+                    options: TuneOptions,
+                    data_extractors: pipeline_dp.DataExtractors,
+                    public_partitions=None) -> TuneResult:
+    """Tunes parameters.
+
+    Args:
+        col: collection where all elements are of the same type.
+          contribution_histograms:
+        options:
+        data_extractors: functions that extract needed pieces of information
+          from elements of 'col'.
+        public_partitions: A collection of partition keys that will be present
+          in the result. If not provided, the tuning for private partition
```

Comment:
Optional: something like `If not provided, tuning will be performed assuming private partition selection is used.` I just want to make it clear that the choice of using private partitions or not affect many other things than just the private partition selection step.